packages/infra/lib/quest/quest-stack.ts

```diff
@@ -0,0 +1,278 @@
+      cors: [
+        {
+          allowedOrigins: ["*"],
+          allowedMethods: [s3.HttpMethods.GET],
+        },
+      ],
+    });
+
+    const { envVars, secrets } = questEnvironmentVariablesAndSecrets({
+      nestedStack: this,
+      quest: props.config.quest,
+      questReplicaBucket: this.questReplicaBucket,
+      labConversionBucket: this.labConversionBucket,
+    });
+
+    const commonConfig = {
+      lambdaLayers: props.lambdaLayers,
+      vpc: props.vpc,
+      envType: props.config.environmentType,
+      sentryDsn: props.config.lambdasSentryDSN,
+      alarmAction: props.alarmAction,
+      quest: props.config.quest,
+      systemRootOID: props.config.systemRootOID,
+      envVars,
+    };
+
+    const sftpAction = this.setupLambda("sftpAction", {
+      ...commonConfig,
+      questReplicaBucket: this.questReplicaBucket,
+      labConversionBucket: this.labConversionBucket,
+    });
+    this.sftpActionLambda = sftpAction.lambda;
+
+    const rosterUpload = this.setupLambda("rosterUpload", {
+      ...commonConfig,
+      questReplicaBucket: this.questReplicaBucket,
+      labConversionBucket: this.labConversionBucket,
+    });
+    this.rosterUploadLambda = rosterUpload.lambda;
+
+    const responseDownload = this.setupLambda("responseDownload", {
+      ...commonConfig,
+      questReplicaBucket: this.questReplicaBucket,
+      labConversionBucket: this.labConversionBucket,
+    });
+    this.responseDownloadLambda = responseDownload.lambda;
+
+    const conversion = this.setupLambda("conversion", {
+      ...commonConfig,
+      questReplicaBucket: this.questReplicaBucket,
+      labConversionBucket: this.labConversionBucket,
+    });
+    this.conversionLambda = conversion.lambda;
+
+    const lambdas = this.getLambdas();
+    for (const secret of secrets) {
+      for (const lambda of lambdas) {
+        secret.grantRead(lambda);
+      }
+    }
+  }
+
+  getLambdas(): Lambda[] {
+    return [
+      this.sftpActionLambda,
+      this.rosterUploadLambda,
+      this.responseDownloadLambda,
+      this.conversionLambda,
+    ];
+  }
+
+  getAssets(): QuestAssets {
+    return {
+      questLambdas: [
+        {
+          envVarName: "QUEST_SFTP_ACTION_LAMBDA_NAME",
+          lambda: this.sftpActionLambda,
+        },
+        {
+          envVarName: "QUEST_ROSTER_UPLOAD_LAMBDA_NAME",
+          lambda: this.rosterUploadLambda,
+        },
+        {
+          envVarName: "QUEST_RESPONSE_DOWNLOAD_LAMBDA_NAME",
+          lambda: this.responseDownloadLambda,
+        },
+        {
+          envVarName: "QUEST_CONVERSION_LAMBDA_NAME",
+          lambda: this.conversionLambda,
+        },
+      ],
+      sftpActionLambda: this.sftpActionLambda,
+      rosterUploadLambda: this.rosterUploadLambda,
+      responseDownloadLambda: this.responseDownloadLambda,
+      conversionLambda: this.conversionLambda,
+      questReplicaBucket: this.questReplicaBucket,
+      labConversionBucket: this.labConversionBucket,
```

Comment:
Don't like that we're mixing the assets and an array of other stuff.

Can we just break it out into something like e.g. getAssets() and getLambdas(), assets returns the assets + lambdas returns the array?

Reply:
Agree - splitting this up.